runtime/mux.go

```diff
@@ -265,7 +270,12 @@ func (s *ServeMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		path = r.URL.RawPath
 	}
 
-	components := strings.Split(path[1:], "/")
+	var components []string
+	if s.unescapingMode == UnescapingModeAllCharacters || s.unescapingMode == UnescapingModeLegacy {
+		components = EncodedPathSplitter.Split(path[1:], -1)
+	} else {
+		components = strings.Split(path[1:], "/")
+	}
```

Comment:
What, if any, are the routing consequences of this change? Is this a change in behavior?